api/saml/logout_request.go

```diff
@@ -53,28 +67,195 @@ func newLogoutRequest(issuer string, username string, sessionIndex string) logou
 	}
 }
 
-// xml returns the xml for the logout request
-func (req *logoutRequest) xml() ([]byte, error) {
-	output, err := xml.MarshalIndent(req, "", "    ")
+func extractCertBody(certFile string) (string, error) {
```

Comment:
I'd replace this with something like this pseudo code:
data, rest := pem.Decode(certFile)
str := base64.StdEncoding.EncodeToString(data)
return str

Reference:
https://golang.org/pkg/encoding/pem/#example_Decode
https://golang.org/pkg/encoding/base64/#Encoding.EncodeToString

Reply:
Yes, that's so much neater. I tried to use that package but gave it up b/c it's encode method includes the headers. I didn't think to encode the bytes myself.